crates/object-store-aws-sdk/src/credential_provider.rs

```diff
@@ -49,35 +88,88 @@ impl S3CredentialProvider {
     ///
     /// Returns an error if the credentials provider cannot be obtained from the SDK configuration.
     pub fn from_config(sdk_config: &SdkConfig) -> object_store::Result<Self> {
-        let credentials =
+        let credentials_provider =
             sdk_config
                 .credentials_provider()
                 .ok_or_else(|| object_store::Error::Generic {
                     store: "S3",
-                    source: "Failed to get S3 credentials from the AWS SDK".into(),
+                    source: "No credentials provider found in SdkConfig".into(),
                 })?;
-
         Ok(Self {
-            credentials: credentials.clone(),
+            cache: IdentityCache::lazy().build(),
+            runtime: Self::build_aws_runtime_components(sdk_config, &Client::new(sdk_config))
+                .map_err(|e| object_store::Error::Generic {
+                    store: "S3",
+                    source: Box::new(e),
+                })?,
+            identity_resolver: SharedIdentityResolver::new(credentials_provider),
         })
     }
+
+    fn build_aws_runtime_components(
+        sdk_config: &SdkConfig,
+        client: &Client,
+    ) -> object_store::Result<RuntimeComponents> {
+        let mut runtime_components = RuntimeComponentsBuilder::new("ServiceRuntimePlugin");
```

Comment:
```suggestion
        let mut runtime_components = RuntimeComponentsBuilder::new("S3CredentialProvider");
```

Shouldn't this name be specific to the component its part of?